src/test/java/org/opensearch/security/hasher/Argon2PasswordHasherTests.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.hasher;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import org.opensearch.security.support.ConfigConstants;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+public class Argon2PasswordHasherTests extends AbstractPasswordHasherTests {
+
+    @Before
+    public void setup() {
+        passwordHasher = new Argon2PasswordHasher(
+            ConfigConstants.SECURITY_PASSWORD_HASHING_ARGON2_MEMORY_DEFAULT,
+            ConfigConstants.SECURITY_PASSWORD_HASHING_ARGON2_ITERATIONS_DEFAULT,
+            ConfigConstants.SECURITY_PASSWORD_HASHING_ARGON2_PARALLELISM_DEFAULT,
+            ConfigConstants.SECURITY_PASSWORD_HASHING_ARGON2_LENGTH_DEFAULT,
+            ConfigConstants.SECURITY_PASSWORD_HASHING_ARGON2_TYPE_DEFAULT,
+            ConfigConstants.SECURITY_PASSWORD_HASHING_ARGON2_VERSION_DEFAULT
+        );
+    }
+
+    @Test
+    public void shouldGenerateValidHashesFromParameters() {
+        PasswordHasher hasher = new Argon2PasswordHasher(47104, 1, 2, 32, "argon2id", 19);
+        String hash = hasher.hash(password.toCharArray());
+        assertThat(hasher.check(password.toCharArray(), hash), is(true));
+        assertThat(hasher.check(wrongPassword.toCharArray(), hash), is(false));
+
+        hasher = new Argon2PasswordHasher(19456, 2, 1, 64, "argon2i", 19);
```

Comment:
This looks like a good contender for using the `@Parameterized` pattern to pass in different parameters for a list of test scenarios. See https://github.com/opensearch-project/security/blob/main/src/test/java/org/opensearch/security/auditlog/impl/AuditCategoryTest.java#L33-L79

This is also fine as is, but I wanted to point out that this repo has examples of passing a list of parameters to run for the same set of assertions.